File: TwoPointers/ReverseString.java
package TwoPointers;

public class ReverseString {
    
    public static void reverseString(char[] s){
        int left = 0;
        int right = s.length - 1;
        char temp;

        while(left<right){
            temp = s[left];
            s[left] = s[right];
            s[right] = temp;
            left++;
            right--;
        }
    }

    public static void main(String[] args){

        char[] s = {'h','e','l','l','o'};
        reverseString(s);
        for ( char c : s){
            System.out.println(c);
        }
    }
    
}

File: TwoPointers/ReverseString.py
class Solution():
    def reverseString(self, s):
        left, right = 0, len(s) -1
        # right = len(s) - 1
        
        while left <= right:
            temp = s[left]
            s[left] = s[right]
            s[right] = temp
            left += 1
            right -= 1

            
            
if __name__ == "__main__":
    s = ["h","e","l","l","o"]
    obj = Solution()
    obj.reverseString(s)
    print(s)